dc/core/MovieUpdater.h

```diff
@@ -82,16 +84,37 @@ class MovieUpdater : public DataSource
     /** Increment and synchronize movie timestamp across all wall processes. */
     bool synchronizeTimestamp( WallToWallChannel& channel );
 
+    /**
+     * Allow the updater to prepare the next frame, i.e. a previous
+     * decode & swap of a movie frame is finished.
+     */
+    void requestNextFrame();
+
+    /**
+     * @return true if after synchronizeTimestamp() the tile needs update aka
+     * decode the next movie frame.
+     */
+    bool updateTile() const;
+
+    /** @return current / max fps, movie position in percentage */
+    QString getStatistics() const;
+
 private:
     MoviePtr _ffmpegMovie;
+    FpsCounter _fpsCounter;
 
+    bool _swapDone;
```

Comment:
I would also rename _swapDone to _nextFrameRequested or similar, because it is set by requestNextFrame(); also, there is not really a concept of "swap" in this class